ecl/hql/hqlutil.cpp

```diff
@@ -443,8 +458,8 @@ IHqlExpression * createPhysicalIndexRecord(HqlMapTransformer & mapper, IHqlExpre
                 Owned<ITypeInfo> hozedType = getHozedKeyType(cur);
                 if (hozedType == cur->queryType())
                     newField = LINK(cur);
-                else if (createKeyedTypes)
-                    newField = createField(cur->queryId(), makeKeyedType(cur->getType()), nullptr, extractFieldAttrs(cur));
+                else if (createKeyedTypes && hozedType->isInteger())
```

Comment:
This was incorrectly mapping ebcdic strings to keyed(ebcdic string) instead of ascii string.  (Keyed renamed to KeyedInt to make the type clear.)